src/extension/sldService/src/test/java/org/geoserver/sldservice/rest/ClassifierTest.java

```diff
@@ -2805,4 +2816,23 @@ public void testClassifyRasterSingleByteClosed() throws Exception {
         assertEntry(entries[0], 10d, null, "#000000", 0);
         assertEntry(entries[1], 10.000000000000002, ">= 10 AND <= 10.000001", "#FF071C", 1);
     }
+
+    @Test
+    public void testClassifyRasterClosedIntervalsNegativeValues() throws Exception {
+        final String restPath =
+                RestBaseController.ROOT_PATH
+                        + "/sldservice/cite:negativeValuesDem/"
+                        + getServiceUrl()
+                        + ".xml?continuous=false&fullSLD=true&method=jenks&intervals=7"
+                        + "&colors=0xFF071C,0xFFA92E&ramp=custom";
+        Document dom = getAsDOM(restPath, 200);
+        print(dom);
+        RasterSymbolizer rs = getRasterSymbolizer(dom);
+        ColorMap cm = rs.getColorMap();
+        ColorMapEntry[] entries = cm.getColorMapEntries();
+        // the max value of the raster is -50. The last entry should have the next
+        // value after it since we are requesting closed intervals
+        ColorMapEntry entry = entries[7];
+        assertEquals(-49.99999999999999, entry.getQuantity().evaluate(null, Double.class), 0d);
```

Comment:
Maybe this test depends too much on the internal implementation that is offsetting your value by a very small amount, I would relax this test a little bit (value between -49.5 and -50 ?)

Reply:
fixed